chats/apps/sectorqueue/models.py

```diff
@@ -0,0 +1,100 @@
+from django.db import models
+from chats.core.models import BaseModel
+from django.utils.translation import gettext_lazy as _
+from chats.apps.sectors.models import Sector, SectorAuthorization
+
+
+class SectorQueue(BaseModel):
+    sector = models.ForeignKey(
+        Sector,
+        verbose_name=_("queues"),
+        related_name="sector_authorizations",
+        on_delete=models.CASCADE,
+        to_field="uuid"
+    )
+    name = models.CharField(_("sector name"), max_length=150, blank=True)
+
+    class Meta:
+        verbose_name = _("Sector Queue")
+        verbose_name_plural = _("Sector Queues")
+
+    def __str__(self):
+        return self.name
+
+    def get_permission(self, user):
+        try:
+            sectorqueue_auth = self.sector.authorizations.get(user=user)
+        except SectorAuthorization.DoesNotExist:
+            if self.sector.project.authorizations.filter(user=user):
+                sectorqueue_auth = True
+            elif self.sector_authorizations.filter(user=user):
+                sectorqueue_auth = True
+            else:
+                sectorqueue_auth = False
+        return sectorqueue_auth
+
+    @property
+    def agent_count(self):
+        return self.sector_authorizations.filter(role=SectorQueueAuthorization.ROLE_AGENT).count()
+
+    def get_or_create_user_authorization(self, user):
+        sector_auth, created = self.sector_authorizations.get_or_create(user=user)
+        return sector_auth
+
+    def set_queue_authorization(self, user, role: int):
+        sector_auth, created = self.sector_authorizations.get_or_create(user=user, role=role)
+        return sector_auth
+
+
+class SectorQueueAuthorization(BaseModel):
```

Comment:
Does not need to reference the sector, call it `QueueAuthorization`. In the same way sectors don't reference project, it's not `ProjectSectorAuthorization` but `SectorAuthorization`

Reply:
I change the model name do QueueAuthorization and remade the migrations, leaving only the initial with the correct name.